testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/javascript/JavascriptAdapterTest.java

```diff
@@ -52,9 +54,12 @@
 import static org.keycloak.testsuite.util.WaitUtils.waitForPageToLoad;
 import static org.keycloak.testsuite.util.WaitUtils.waitUntilElement;
 
-/**
- * @author mhajas
- */
+@AppServerContainer(ContainerConstants.APP_SERVER_WILDFLY)
+@AppServerContainer(ContainerConstants.APP_SERVER_WILDFLY_DEPRECATED)
+@AppServerContainer(ContainerConstants.APP_SERVER_EAP)
+@AppServerContainer(ContainerConstants.APP_SERVER_EAP6)
+@AppServerContainer(ContainerConstants.APP_SERVER_UNDERTOW)
+@AppServerContainer(ContainerConstants.APP_SERVER_EAP71)
```

Comment:
This change has been specifically introduced for my Pull Request against the pipeline. Please have a quick glance at the before reading below.

Before this change, the Javascript adapter tests were invoked with each App Server. This was absolutely fine, since we tested only Wildfly App Server in the pipeline. However, with matrix-like approach to the adapter tests (we execute the adapter tests with different App Servers such as Wildfly, Jetty 9.2, Jetty 9.3 etc) this becomes wasteful (especially in terms of testing time). 

While working on this, I wasn't sure with what App Servers combinations we want to run Javascript adapter tests. I only knew, they don't make sense (and probably nobody expects them to run) with Jetty. That's why there are all App Servers on the list except the new one - Jetty. 

With your explanation, I think we can enhance this solution a bit more. Perhaps Javascript Adapter test should only run with the default App Server (Undertow). However, we would also need to add Undertow to the pipeline. If you agree with me, I would like to create a JIRA for it. We are approaching to the end of the sprint and my Pull Request against the pipeline hasn't been reviewed so far, so it's a bit risky to do this change now.

Alternatively, if you don't like this approach, I can remove all the annotations and allow this test to execute always. But as I mentioned before, it seems a bit wasteful to me.

Reply:
I see you are right. Probably we should run it only with undertow, however, it might be a little bit confusing for others looking at it and say "why it only runs with undertow, do they even test Javascript adapter with wildfly?". 

Maybe we could create a separate package for adapter tests which can run without application server (maybe simply javascript package). This way we won't have the problem of multiple runs as it will be part of base testsuite. @slaskawi @vramik WDYT?

Reply:
Probably the best option would be to create a new, special group for Javascript (and possibly other, non-app-server tests). I created [KEYCLOAK-9712](https://issues.jboss.org/browse/KEYCLOAK-9712) to implement this. 

Since it's getting late in this sprint, let me revert this change to Javascript tests (and let them always execute). We can revisit this thread once working on KEYCLOAK-9712.

Reply:
ok I agree